src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,84 @@
+package duke;
+
+import java.util.Scanner;
+
+public class Duke {
+
+    private Scanner getInput;
+    private Ui ui;
+    private Storage storage;
+    private TaskList taskList;
+
+    private Duke(String filePath) {
+        this.getInput = new Scanner(System.in);
+        this.ui = new Ui();
+        this.storage = new Storage(filePath);
+        this.taskList = new TaskList(storage.loadFile());
+
+        ui.printIntro();
+        ui.printList(taskList);
+    }
+
+    private String askForInput() {
+        System.out.print("> ");
+        return getInput.nextLine();
+    }
+
+    private void run() {
+        // Execute inputs
+        String userInput;
+        EventType curEvent;
+        loop: while (true) {
+
+            // Get inputs
+            userInput = askForInput();
+
+            // Check if valid
+            try {
+                curEvent = Parser.parse(userInput);
+            } catch (DukeException e) {
+                System.out.println(e.getMessage());
+                continue; // skip the rest of the code and go to next iteration
+            }
+
+            // If valid keyword, execute input
+            switch (curEvent) {
+                case BYE:
```

Comment:
Java coding standard we follow suggest omitting indentation for case clauses. Maybe we can remove those?